src/main/java/seedu/address/model/student/Student.java

```diff
@@ -122,19 +125,32 @@ public String toString() {
      *
      * @param date The date on which attendance is being recorded.
      * @param status The attendance status, either 'present' or 'absent'.
-     * @throws IllegalArgumentException if the provided status is invalid.
+     * @throws IllegalArgumentException if the provided status is nvalid.
      */
     public void markAttendance(LocalDate date, String status) {
-        PersonAttendance attendance = new PersonAttendance(status);
-        attendanceRecords.put(date, attendance);
+        Attendance attendance = new Attendance(status);
+        AttendanceRecord record = new AttendanceRecord(date, attendance);
+        attendanceRecords.add(record);
```

Comment:
This adds an AttendanceRecord for every new marking of attendance even if it is for the same date. The GetAttendanceCommand also only gets the first attendance record it finds that has the same date, so it ignores the later attendance that has been marked.
Maybe we can overwrite the old record if we find an AttendanceRecord for the same date in the list.